### PR TITLE
[OpenAPI-gen] Add Update and Delete endpoints

### PR DESCRIPTION
**Pull request by c-harding**

### CODE REVIEW

**Review by willjvsmith — APPROVED** [review]
LGTM 🚀 

Probs wanna make the endpoints generate in CRUD order to be consistent. Also I'm not spotting the `required` field anywhere, is it not recommended?

File: src/test/scala/temple/generate/target/openapi/OpenAPIGeneratorTest.scala [hunk 1]
```diff
@@ -138,11 +140,39 @@ class OpenAPIGeneratorTest extends FlatSpec with Matchers {
...
+        |      responses:
+        |        '500':
+        |          $ref: '#/components/responses/Error500'
+        |        '404':
+        |          $ref: '#/components/responses/Error404'
+        |        '401':
+        |          $ref: '#/components/responses/Error401'
+        |        '400':
+        |          $ref: '#/components/responses/Error400'
+        |        '200':
```
Comment (jaylees14) [inline src/test/scala/temple/generate/target/openapi/OpenAPIGeneratorTest.scala]: Previous responses are in increasing order (200, 400, 401, 500), whereas these are in decreasing. Should we be consistent?

**Review by jaylees14 — APPROVED** [review]
LGTM 💯

File: src/test/scala/temple/generate/target/openapi/OpenAPIGeneratorTest.scala (hunk 1 above)
Comment (c-harding) [reply src/test/scala/temple/generate/target/openapi/OpenAPIGeneratorTest.scala]: They’re in decreasing order purely as a side-effect of the use of unordered maps. Sorting that out now

**Comment by c-harding** [comment]
> Also I'm not spotting the `required` field anywhere, is it not recommended?

@smithwjv next PR

**Review by jaylees14 — APPROVED** [review]
Changes look good, thanks @xsanda 😄

**Pull request merged**